lib/proxy/plugin_initializer.rb

```diff
@@ -1,36 +1,288 @@
-class ::Proxy::PluginInitializer
-  def initialize_plugins(loaded_plugins)
-    instantiated_plugins = instantiate_plugins(loaded_plugins)
-    in_configuration_order = build_configuration_order(instantiated_plugins)
-    configure_plugins(in_configuration_order, instantiated_plugins)
+require 'proxy/default_di_wirings'
+require 'proxy/default_plugin_validators'
+
+class ::Proxy::PluginGroup
+  include ::Proxy::Log
+  attr_reader :plugin, :providers, :state, :di_container
+
+  def initialize(a_plugin, providers = [], di_container = ::Proxy::DependencyInjection::Container.new)
+    @plugin = a_plugin
+    @state = :starting
+    @providers = providers
+    @di_container = di_container
+  end
+
+  def failed?
+    @state == :failed
+  end
+
+  def resolve_providers(all_plugins_and_providers)
+    return if failed?
+    return unless @plugin.uses_provider?
+
+    used_providers = [@plugin.settings.use_provider].flatten.map(&:to_sym)
+    providers = all_plugins_and_providers.select {|p| used_providers.include?(p[:name].to_sym)}
+
+    not_available = used_providers - providers.map {|p| p[:name].to_sym}
+    return @providers = providers.map {|p| p[:class]} if not_available.empty?
+
+    fail_group_with_message("Disabling all modules in the group '#{member_names}': following providers are not available #{not_available}")
```

Comment:
`member_names` here should be the group name?  Or should it be a list of members - if so, it should probably be printed more clearly with a join(", ") or something, as it appears as:

    Disabling all modules in the group '[:hello]' due